pkg/bgpv1/manager/reconcilerv2/service.go

```diff
@@ -0,0 +1,535 @@
+		desiredSvcPaths, err = r.svcDiffReconciliation(p, ls)
+		if err != nil {
+			return err
+		}
+	}
+
+	var allErr error
+	for svc, desiredAFPaths := range desiredSvcPaths {
+		metadata := r.getMetadata(p.BGPInstance)
+
+		// check if service exists
+		currentAFPaths, exists := metadata.ServicePaths[svc]
+		if !exists && len(desiredAFPaths) == 0 {
+			// service does not exist in our local state, and there is nothing to advertise
+			continue
+		}
+
+		// reconcile service paths
+		updatedAFPaths, err := ReconcileAFPaths(&ReconcileAFPathsParams{
+			Logger:       r.logger.WithField(types.InstanceLogField, p.DesiredConfig.Name),
+			Ctx:          ctx,
+			Instance:     p.BGPInstance,
+			DesiredPaths: desiredAFPaths,
+			CurrentPaths: currentAFPaths,
+		})
+
+		if err == nil && len(desiredAFPaths) == 0 {
+			// no error is reported and desiredAFPaths is empty, we should delete the service
+			delete(metadata.ServicePaths, svc)
+		} else {
+			// update service paths with returned updatedAFPaths even if there was an error.
+			metadata.ServicePaths[svc] = updatedAFPaths
+		}
+
+		r.setMetadata(p.BGPInstance, metadata)
+		allErr = errors.Join(allErr, err)
+	}
+
+	return allErr
+}
+
+// modifiedServiceAdvertisements compares local advertisement state with desiredPeerAdverts, if they differ, it updates the local state and returns true
+// for full reconciliation.
+func (r *ServiceReconciler) modifiedServiceAdvertisements(p ReconcileParams, desiredPeerAdverts PeerAdvertisements) bool {
+	// current metadata
+	serviceMetadata := r.getMetadata(p.BGPInstance)
+
+	// check if BGP advertisement configuration modified
+	modified := !PeerAdvertisementsEqual(serviceMetadata.ServiceAdvertisements, desiredPeerAdverts)
+
+	// update local state, if modified
+	if modified {
+		r.setMetadata(p.BGPInstance, ServiceReconcilerMetadata{
+			ServicePaths:          serviceMetadata.ServicePaths,
+			ServiceAdvertisements: desiredPeerAdverts,
+		})
+	}
+
+	return modified
+}
+
+// Populate locally available services used for externalTrafficPolicy=local handling
+func (r *ServiceReconciler) populateLocalServices(localNodeName string) (sets.Set[resource.Key], error) {
+	ls := sets.New[resource.Key]()
+
+	epList, err := r.epDiffStore.List()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list upsertEPs from diffstore: %w", err)
+	}
+
+endpointsLoop:
+	for _, eps := range epList {
+		_, exists, err := r.resolveSvcFromEndpoints(eps)
+		if err != nil {
+			// Cannot resolve service from upsertEPs. We have nothing to do here.
+			continue
+		}
+
+		if !exists {
+			// No service associated with this endpoint. We're not interested in this.
+			continue
+		}
+
+		svcKey := resource.Key{
+			Name:      eps.ServiceID.Name,
+			Namespace: eps.ServiceID.Namespace,
+		}
+
+		for _, be := range eps.Backends {
+			if be.NodeName == localNodeName {
+				// At least one endpoint is available on this node. We
+				// can make unavailable to available.
```

Comment:
not sure what is the meaning of "We an make unavailable to available"?

Reply:
Fixed it - left overs from reconciler code.